engine/src/main/scala/cromwell/webservice/WorkflowSubmitBatchActor.scala

```diff
@@ -0,0 +1,47 @@
+package cromwell.webservice
+
+import akka.actor.{Actor, ActorRef, Props}
+import cromwell.engine.WorkflowSourceFiles
+import cromwell.engine.workflow.WorkflowManagerActor
+import cromwell.webservice.CromwellApiHandler.{WorkflowManagerBatchSubmitResponse, WorkflowManagerResponse}
+
+/** Submits a sequence of sources, then messages back the respective sequence of responses. */
+class WorkflowSubmitBatchActor(responseActor: ActorRef, requestHandlerActor: ActorRef,
+                               sources: Seq[WorkflowSourceFiles]) extends Actor {
+
+  private val responses = Array.fill[Option[WorkflowManagerResponse]](sources.size)(None)
+
+  override def preStart() = {
+    sources.zipWithIndex foreach {
+      case (source, index) =>
+        context.actorOf(
+          Props(new WorkflowSubmitIndexedActor(self, requestHandlerActor, source, index)),
+          "WorkflowSubmitIndexedActor_" + index)
+    }
```

Comment:
Nitpick: If this is done in the `preStart` then I think it's equivalent to messaging directly the WorkflowManagerActor from here instead of creating an actor to do it, and then handling the response in this actor too, because it won't start processing new messages before `preStart` is done.